src/mame/act/victor9k.cpp

```diff
@@ -679,27 +679,26 @@ void victor9k_state::machine_reset()
 void victor9k_state::victor9k(machine_config &config)
 {
 	// basic machine hardware
-	I8088(config, m_maincpu, XTAL(30'000'000)/6);
+	I8088(config, m_maincpu, 5_MHz_XTAL);
 	m_maincpu->set_addrmap(AS_PROGRAM, &victor9k_state::victor9k_mem);
 	m_maincpu->set_irq_acknowledge_callback(I8259A_TAG, FUNC(pic8259_device::inta_cb));
 
 	// video hardware
+	int char_width = 10;
 	screen_device &screen(SCREEN(config, SCREEN_TAG, SCREEN_TYPE_RASTER));
-	screen.set_color(rgb_t::green());
-	screen.set_refresh_hz(50);
-	screen.set_vblank_time(ATTOSECONDS_IN_USEC(2500)); // not accurate
+	screen.set_raw(29.4912_MHz_XTAL/ char_width, 1200, 0, 801, 550, 0, 410);
```

Comment:
This can't be correct. I assume you mean just `29.4912_MHz_XTAL` for the dot clock. Are the other values based on writes to the 6845?

Reply:
According to the schematics, the master XTAL is 15.0MHz.
(CPU clock is also from there btw)

Reply:
Ok, I tested with the value of 
`screen.set_raw(**15_MHz_XTAL**, 1200, 0, 801, 550, 0, 410);`
and no other changes. When I boot the computer it's in low resolution mode, works fine. If I then enter high resolution mode, all good so far. When I exit the program that's using high resolution and return to low resolution I get a bunch of errors:
> AGX: Texture read/write assertion failed: bytes_per_row >= used_bytes_per_row
> AGX: Texture read/write assertion failed: bytes_per_row >= used_bytes_per_row
This happens in both MS-DOS and CP/M-86. For some reason that doesn't happen when I use the 29.4912_MHz_XTAL/ char_width clock value. I know the high resolution mode has 16 pixels / width instead of 10. Was the incorrect resolution just hiding some other bug related to the pixel density?

Reply:
I mean, the master XTAL is 15MHz (in other words, there is no 30MHz or imaginary 29.xxxMHz or 5MHz XTAL). Everything is derived from the master XTAL.
cpu clock is 15/3
VIA clock is 15/3/5
CRT clock is 15->stuff->crtc

Reply:
Ignore that last comment, I'm realizing I missed a call to reset the clock that wasn't needed. The code was changing the clock on exiting the hires mode.

Reply:
On the 1200 and 801 values. 

For 1200: 
> // param [in] htotal Total pixel clocks per line, including horizontal blanking period.

I put in a guess of what the total would be. I understood the expected value to be <width of visible screen> + <horizontal blanking period>. Width of screen is 800, I know the value of the monitors states the blanking period needs to be 1200µsec at minimum. I didn't understand how to translate µsec to the expected value here. Can you recommend a better value? I wasn't sure how to derive the correct number.

For the 801:
> //param [in] hbstart Index of first pixel in horzontal blanking period after visible pixels.
The screen is 800 pixels wide, so 801 would definitely be the first pixel after the visible pixels and would start the blanking period. Am I understanding this comment incorrectly?

Reply:
You don't need to guess. The machine will write the expected screen parameters to the 6845 CRTC. Basically you need to turn on logging for it and see what the machine writes.

As for the formula to calculate the values, you can find it in `screen.h`.

Reply:
NOTE: my bad: the dot clock is not the 15MHz master XTAL, but what appears to be an RC clock, 17.5MHz
800*400 wouldn't be possible with 15MHz so I double checked.

Reply:
Thanks for the tip that these values are logged. I think I have the values for the standard resolution and the high resolution.

> [:11a] **M6845 config screen: HTOTAL: 1488  VTOTAL: 422  MAX_X: 1279  MAX_Y: 399  HSYNC: 1296-1487  VSYNC: 400-411  Freq: 382.204556fps**
> [:11a] ':8l' (3CF53):M6845 reg 0x0a = 0x20
> [:screen] Partial: update_partial(:screen, 409): [:screen] updating 0-409
> [:11a] M6845: Valid screen parameters - display reenabled!!!
> [:11a] **M6845 config screen: HTOTAL: 930  VTOTAL: 422  MAX_X: 799  MAX_Y: 399  HSYNC: 810-929  VSYNC: 400-411  Freq: 382.204556fps**

I see now part of my confusion is the MAX_X value and the HTOTAL values are quite different depending on which resolution is in play.

Reply:
382.20 fps is wrong tho, if the clock source is really 15 MHz then dividing it by 8 should give back a more sane 47.775 fps (is it tho?)

Reply:
The hardware manual states:

> The horizontal sweep rate is approximately 15KHz. A vertical refresh rate of 76 Hz, or. 76 frames per second, prevents visual flicker.
When I set the pixel clock to 15 MHz / 3 / 2 as you suggest I get  Freq: 63.7fps which is certainly more in the ballpark. If I set the pixel clock to 15MHz / 5 I get 76.4fps, which is spot on to the manual. Is that how I should derive the pixel clock, by reversing out what's giving the correct FPS? 

I'm a newbie at reading schematics and running through what the 74LS163 is doing between the 15MHz input clock to the CRTCLK output on the schematics is a bit over my head.

In unrelated news. I also cleaned up the logging to conform to what looks more modern in the code. I also broke out the resolution setting functionality into the crtc_begin_update callback instead of happening at the top of crtc_update_row callback. That seemed cleaner to me.

Reply:
Ok, I removed the set_raw() and went back to the earlier method of setting the parameters.